Should use standard '@typescript-eslint/' prefix for rule names, not 'ts/'

### Describe the bug

The new ESLint flat config allows assigning any arbitrary prefix you want for rule names. The prefix used in the `plugins` object:

https://github.com/TanStack/config/blob/2935da96712b917efd39f06d440998a5a8c49e1b/packages/config/src/eslint/index.js#L44-L46

...is then what the rules are referred to as in `rules`:

https://github.com/TanStack/config/blob/2935da96712b917efd39f06d440998a5a8c49e1b/packages/config/src/eslint/typescript.js#L4-L6



### Steps to reproduce

1. `import { tanstackConfig } from '@tanstack/config/eslint'` in an `eslint.config.js`
2. See that rules have names like `ts/array-type`, not `@typescript-eslint/array-type`

### Expected behavior

The [recommended preset configs from typescript-eslint](https://typescript-eslint.io/users/configs) use a `'@typescript-eslint/'` prefix, not `'ts/'`. So if you try to, say, use both `@tanstack/config/eslint` and `typescript-eslint` in a config:

```js
import { tanstackConfig } from '@tanstack/config/eslint'
import tseslint from 'typescript-eslint';

export default [
  ...tanstackConfig,
  { 
    rules: {
      'ts/no-explicit-any': 'error', // (just as an example)
    },
  },
  ...tseslint.configs.recommended,
]
```

...you'll get each rule running and reporting _twice_:

```plaintext
$ pnpm run test:eslint

...

   ✖  nx run @tanstack/form-core:test:eslint
      > @tanstack/form-core@0.25.2 test:eslint /Users/josh/repos/tanstack-form/packages/form-core
      > eslint ./src ./tests
      
      
      /Users/josh/repos/tanstack-form/packages/form-core/src/FieldApi.ts
        418:13  error  Unexpected any. Specify a different type  ts/no-explicit-any
        418:13  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
      ...
```

Change request: can we switch the prefix to `'@typescript-eslint/'`? 

### How often does this bug happen?

Every time

### Screenshots or Videos

_No response_

### TanStack Config version

0.9.2

### TypeScript version

_No response_

### Additional context

I realize the `ts/` prefix is _much easier_ to read+write than the super verbose `@typescript-eslint/`. But we're stuck with `@typescript-eslint/` in typescript-eslint because that's what everyone's been using for years. 

We in typescript-eslint have previously surfaced feedback to ESLint core that prefix duplication is a known user pain: https://github.com/eslint/eslint/discussions/17766. But there isn't a way to lock down and/or deduplicate the prefixes (yet?).

Keeping to the standard prefix makes it easier to use standard shared configs: https://github.com/TanStack/config/discussions/114.

--- a/packages/config/src/eslint/index.js
+++ b/packages/config/src/eslint/index.js
@@ -43,7 +43,7 @@ export const tanstackConfig = [
     },
     plugins: {
       // @ts-expect-error
-      ts: tseslint.plugin,
+      '@typescript-eslint': tseslint.plugin,
       // @ts-expect-error
       import: pluginImport,
       node: pluginNode,

--- a/packages/config/src/eslint/typescript.js
+++ b/packages/config/src/eslint/typescript.js
@@ -3,9 +3,9 @@
 /** @type {import('eslint').Linter.RulesRecord} */
 export const typescriptRules = {
   /** Prefer Array<T> format */
-  'ts/array-type': ['error', { default: 'generic', readonly: 'generic' }],
+  '@typescript-eslint/array-type': ['error', { default: 'generic', readonly: 'generic' }],
   /** Prevent @ts-ignore, allow @ts-expect-error */
-  'ts/ban-ts-comment': [
+  '@typescript-eslint/ban-ts-comment': [
     'error',
     {
       'ts-expect-error': false,
@@ -13,13 +13,13 @@ export const typescriptRules = {
     },
   ],
   /** Bans problematic built-in types and suggests alternatives */
-  'ts/ban-types': 'error',
+  '@typescript-eslint/ban-types': 'error',
   /** Enforce import type { T } */
-  'ts/consistent-type-imports': ['error', { prefer: 'type-imports' }],
+  '@typescript-eslint/consistent-type-imports': ['error', { prefer: 'type-imports' }],
   /** Shorthand method style is less strict */
-  'ts/method-signature-style': ['error', 'property'],
+  '@typescript-eslint/method-signature-style': ['error', 'property'],
   /** Enforces generic type convention */
-  'ts/naming-convention': [
+  '@typescript-eslint/naming-convention': [
     'error',
     {
       selector: 'typeParameter',
@@ -33,29 +33,29 @@ export const typescriptRules = {
     },
   ],
   /** Duplicate values can lead to bugs that are hard to track down */
-  'ts/no-duplicate-enum-values': 'error',
+  '@typescript-eslint/no-duplicate-enum-values': 'error',
   /** Using the operator any more than once does nothing */
-  'ts/no-extra-non-null-assertion': 'error',
+  '@typescript-eslint/no-extra-non-null-assertion': 'error',
   /** There are several potential bugs with this compared to other loops */
-  'ts/no-for-in-array': 'error',
+  '@typescript-eslint/no-for-in-array': 'error',
   /** Enforce valid definition of new and constructor */
-  'ts/no-misused-new': 'error',
+  '@typescript-eslint/no-misused-new': 'error',
   /** Disallow TypeScript namespaces */
-  'ts/no-namespace': 'error',
+  '@typescript-eslint/no-namespace': 'error',
   /** Disallow non-null assertions after an optional chain expression */
-  'ts/no-non-null-asserted-optional-chain': 'error',
+  '@typescript-eslint/no-non-null-asserted-optional-chain': 'error',
   /** Detects conditionals which will always evaluate truthy or falsy */
-  'ts/no-unnecessary-condition': 'error',
+  '@typescript-eslint/no-unnecessary-condition': 'error',
   /** Checks if the the explicit type is identical to the inferred type */
-  'ts/no-unnecessary-type-assertion': 'error',
+  '@typescript-eslint/no-unnecessary-type-assertion': 'error',
   /** Don't over-define types for simple things like strings */
-  'ts/no-inferrable-types': ['error', { ignoreParameters: true }],
+  '@typescript-eslint/no-inferrable-types': ['error', { ignoreParameters: true }],
   /** Enforce the use of as const over literal type */
-  'ts/prefer-as-const': 'error',
+  '@typescript-eslint/prefer-as-const': 'error',
   /** Prefer for-of loop over the standard for loop */
-  'ts/prefer-for-of': 'warn',
+  '@typescript-eslint/prefer-for-of': 'warn',
   /** Warn about async functions which have no await expression */
-  'ts/require-await': 'warn',
+  '@typescript-eslint/require-await': 'warn',
   /** Prefer of ES6-style import declarations */
-  'ts/triple-slash-reference': 'error',
+  '@typescript-eslint/triple-slash-reference': 'error',
 }

